delivery_price_method/models/delivery_carrier.py

```diff
@@ -24,10 +24,10 @@ def rate_shipment(self, order):
         previous_method = False
         if self.price_method in ("fixed", "base_on_rule"):
             previous_method = self.delivery_type
-            self.delivery_type = self.price_method
+            self.sudo().delivery_type = self.price_method
```

Comment:
```suggestion
            _self = self
            if not _self.check_access_rights("write"):
                _self = _self.sudo()
           _self.delivery_type = self.price_method
```

Reply:
Although it can avoid sudo in v12 where the uid is changed, I don't think in this specific case it's going to add too much.